framework/auth/views.py

```diff
@@ -564,6 +546,25 @@ def external_login_confirm_email_get(auth, uid, token):
             can_change_preferences=False,
         )
 
+    # if user is already logged in
+    if auth and auth.user:
+        # if it is a wrong user
+        if auth.user._id != user._id:
+            return auth_logout(redirect_url=request.url)
+        # if it is the expected user
+        new = request.args.get('new', None)
+        if destination in campaigns.get_campaigns():
+            # external domain takes priority
+            campaign_url = campaigns.external_campaign_url_for(destination)
+            if not campaign_url:
+                campaign_url = campaigns.campaign_url_for(destination)
+            return redirect(campaign_url)
+        if new:
+            status.push_status_message(language.WELCOME_MESSAGE, kind='default', jumbotron=True, trust=True, id='welcome_message')
+        return redirect(web_url_for('dashboard'))
+
+    user.date_last_logged_in = timezone.now()
```

Comment:
Why was this moved? It looks like this change will never get saved.

Reply:
I'm not sure what you mean, the idea with this is that if the user was logged in already their `date_last_logged_in` shouldn't change, but if they log in with this function it should miss the redirect and hit that line.

Reply:
OK, but the user isn't being saved after this point.